src/pages/en/guides/deploy/cloudflare.md

```diff
@@ -5,30 +5,109 @@ layout: ~/layouts/DeployGuideLayout.astro
 
-3. Click the **Save and Deploy** button.
+7. Click the **Save and Deploy** button.
+
+## How to deploy a site using Wrangler
+
+1. Install [Wrangler CLI](https://developers.cloudflare.com/workers/wrangler/get-started/).
+2. Authenticate Wrangler with your Cloudflare account using `wrangler login`.
+3. Run your build command.
+4. Deploy using `npx wrangler pages publish dist`.
+
+```bash
+# Install Wrangler CLI
+npm install -g wrangler
+# Login to Cloudflare account from CLI
+wrangler login
+# Run your build command
+npm run build
+# Create new deployment
+npx wrangler pages publish dist
+```
+
+After your assets are uploaded, Wrangler will give you a preview URL to inspect your site. When you log into the Cloudflare Pages dashboard, you will see your new project.
+
+### Enabling Preview locally with Wrangler
+
+For the preview to work, you must install `wrangler`
+
+```bash
+pnpm install wrangler --save-dev
+```
+
+It's then possible to update the preview script in your `package.json` to `"preview": "wrangler pages dev ./dist"`
 
 ## How to deploy an SSR site
 
 You can also deploy an Astro SSR site to Cloudflare Pages using the [`@astrojs/cloudflare` adapter](https://github.com/withastro/astro/tree/main/packages/integrations/cloudflare#readme).
 
+To enable an SSR site and deploy to Cloudflare Pages, you will need to:
+
+1. Add the [`@astrojs/cloudflare` adapter](https://github.com/withastro/astro/tree/main/packages/integrations/cloudflare#readme) to your project's `package.json` by running:
+
+```bash
+npm install --save-dev @astrojs/cloudflare
+```
+
+2. Add the following to your `astro.config.js` file:
+
+```diff
+import { defineConfig } from 'astro/config';
++ import cloudflare from '@astrojs/cloudflare';
+
+export default defineConfig({
++  output: 'server',
++  adapter: cloudflare()
+});
+```
+
+### Modes
+
+There are currently two modes supported when using Pages Functions with the [`@astrojs/cloudflare`](https://github.com/withastro/astro/tree/main/packages/integrations/cloudflare#readme) adapter. 
+
+1. **Advanced** mode: This mode is used when you want to run your function in `advanced` mode which picks up the `_worker.js` in `dist`, or a directory mode where pages will compile the worker out of a functions folder in the project root.  
+
+> If no mode is set, the default  is `"advanced"`
+
+2. **directory** mode: This mode is used when you want to run your function in `directory` mode, which means the adapter will compile the client side part of you app the same way, but it will move the worker script into a `functions` folder in the project root. The adaptor will only ever place a `[[path]].js` in that folder, allowing you to add additional plugins and pages middleware which can be checked into version control.
+
+```ts
+// directory mode
+export default defineConfig({
+  adapter: cloudflare({ mode: "directory" }),
+});
+```
```

Comment:
``````suggestion
```ts title="astro.config.mjs" "directory"
export default defineConfig({
  adapter: cloudflare({ mode: "directory" }),
});
```
``````
We also have syntax so that we can highlight just the word "directory" so I've done that instead of the code comment. Let's see how it looks!